### PR TITLE
Openshift machine api

### PR DESCRIPTION
**Pull request by enxebre**
We are transitioning the cluster API types from k8s.io over a machine.openshift.io group.
https://github.com/openshift/cluster-api/tree/91fca585a85b163ddfd119fd09c128c9feadddca
Although this give us a necessary level of autonomy today, we aim to consolidate back with upstream once there's a release of the API stable enough
For reference:
Backward compatible changes on machine api operator
https://github.com/openshift/machine-api-operator/pull/185
https://github.com/openshift/machine-api-operator/pull/188
https://github.com/openshift/machine-api-operator/pull/191
AWS actuator support for new group
https://github.com/openshift/cluster-api-provider-aws/pull/147

### CODE REVIEW

**Comment by jwforres** [comment]
cc @spadgett since this impacts the console

**Comment by spadgett** [comment]
Are there any API changes other than API group and version?

WIP console PR at https://github.com/openshift/console/pull/1161

**Comment by enxebre** [comment]
friendly ping @abhinavdahiya @wking :)

File: pkg/asset/machines/master.go [hunk 1]
```diff
@@ -144,7 +145,20 @@ func masterPool(pools []types.MachinePool) types.MachinePool {
 	return types.MachinePool{}
 }
 
-func listFromMachines(objs []clusterapi.Machine) *metav1.List {
+func listFromMachines(objs []machineapi.Machine) *metav1.List {
```
Comment (abhinavdahiya) [inline pkg/asset/machines/master.go]: why this change.. ?

Comment (enxebre) [reply pkg/asset/machines/master.go]: becomes machines now belong to machine.openshift.io API group, hence the `machineapi.Machine` type

**Comment by enxebre** [comment]
openstack:
`level=fatal msg="failed to fetch Terraform Variables: failed to load asset \"Install Config\": invalid \"install-config.yaml\" file: [platform.openstack.region: Unsupported value: \"RegionOne\": supported values: \"regionOne\", platform.openstack.computeFlavor: Unsupported value: \"\": supported values: \"m1.large\", \"m1.large2\", \"m1.tiny\", \"m1.small\", \"m1.medium\", \"m1.xlarge\"]"
`
/test e2e-libvirt
/test e2e-aws
libvirt:
`error: Missing or incomplete configuration info.  Please login or point to an existing, complete config file:`

**Comment by flaper87** [comment]
@enxebre the openstack job should be optional. We're aware that it's broken. Sadly, we need to move this job to another cloud and we're working on getting such cloud ready. :smile:

**Comment by derekwaynecarr** [comment]
@flaper87 - given it’s broken, any objection to merging this and have openstack make similar update at a later date as part of restoring the job behavior?

**Comment by trown** [comment]
WRT openstack: I can manually test against my personal vexxhost tenant.

**Comment by enxebre** [comment]
@trown thanks, just to clarify this openstack/libvirt fail are unrelated to this PR. As discussed with @flaper87 we are providing a backward compatible pivoting path for any actuator https://github.com/openshift/machine-api-operator/pull/192 Once there's a PR analogous to this one for Opensatck we will fully drop support for `cluster.k8s.io` on the machine api operator https://github.com/openshift/machine-api-operator/pull/195

File: pkg/asset/machines/master.go (hunk 1 above)
Comment (abhinavdahiya) [reply pkg/asset/machines/master.go]: Sorry wrong line, more as to why keep the deprecated one too? It that bring used?

**Comment by trown** [comment]
@enxebre awesome, I can work on the follow-up PR for openstack

File: pkg/asset/machines/master.go (hunk 1 above)
Comment (enxebre) [reply pkg/asset/machines/master.go]: @abhinavdahiya we are providing a backward compatible pivoting path for any actuator [openshift/machine-api-operator#192](https://github.com/openshift/machine-api-operator/pull/192) This PR finalise the pivot for aws. We'll put a follow up for libvirt right away and once the openstack actuator is updated to rely on machine.openshift.io and they put a PR analogous to this one for Opensatck we will fully drop support for `cluster.k8s.io` either here and on the machine api operator [openshift/machine-api-operator#195](https://github.com/openshift/machine-api-operator/pull/195)

**Comment by abhinavdahiya** [comment]
Looks good.
/approve

**Comment by trown** [comment]
I think this may actually break openstack. Could we wait a bit to merge while I look into it?

**Comment by derekwaynecarr** [comment]
@trown can you provide an estimate on how long you will need?  this will block our ability to make follow-on changes required for beta 2 and GA on 4.0, and I want to make sure we pro-actively find issues and resolve them as soon as possible.

**Comment by trown** [comment]
@derekwaynecarr I was able to confirm that this patch does break openstack. In particular it is the network/cluster bits that do not add backwards-compatibility.  I tested with a revert of just those bits[1] and it succeeded.

[1] https://github.com/trown/installer/commit/dc5059d0d012ea8c31dae560d5dd8668ce177134

**Comment by enxebre** [comment]
@flaper87 @trown so is the open stack actuator relying on the cluster object?
I can just update the PR to leave the cluster object untouched

**Comment by tomassedovic** [comment]
@enxebre @trown I don't have a great visibility into the openstack actuator yet, but I can confirm that this PR as is breaks openstack and @trown's suggestion fixes it for me.

Hopefully @flaper87 can be more helpful.

**Comment by trown** [comment]
I did a second run against vexxhost with just this patch (not my "fix"), and got the same result:

$ oc logs clusterapi-manager-controllers-5c9cd779b-c2mpx -n openshift-cluster-api -c machine-controller
I0205 18:30:44.238851       1 controller.go:114] Running reconcile Machine for trown-ef260-master-0
E0205 18:31:14.357520       1 controller.go:169] Error checking existence of machine instance for machine object trown-ef260-master-0; Create providerClient err: Post https://auth.vexxhost.net/v3/auth/tokens: dial tcp: i/o timeout
I0205 18:31:15.360038       1 controller.go:114] Running reconcile Machine for trown-ef260-master-1
E0205 18:31:45.701125       1 controller.go:169] Error checking existence of machine instance for machine object trown-ef260-master-1; Create providerClient err: Post https://auth.vexxhost.net/v3/auth/tokens: dial tcp: i/o timeout
I0205 18:31:46.701641       1 controller.go:114] Running reconcile Machine for trown-ef260-master-2
E0205 18:32:16.730833       1 controller.go:169] Error checking existence of machine instance for machine object trown-ef260-master-2; Create providerClient err: Post https://auth.vexxhost.net/v3/auth/tokens: dial tcp: i/o timeout
I0205 18:32:17.731391       1 controller.go:114] Running reconcile Machine for trown-ef260-worker-wpf5b
E0205 18:32:47.775519       1 controller.go:169] Error checking existence of machine instance for machine object trown-ef260-worker-wpf5b; Create providerClient err: Post https://auth.vexxhost.net/v3/auth/tokens: dial tcp: i/o timeout

**Comment by enxebre** [comment]
@flaper87 @trown I updated the PR to keep the cluster object so the OpenStack actuator is happy.
I'd suggest we move forward this PR and do any farther follow up fix based on CI, blocking PRs on manual runs of the cluster is not a good procedure for anyone :)

**Comment by derekwaynecarr** [comment]
@enxebre agreed.  we need to move this forward if we have a green CI signal.

**Comment by tomassedovic** [comment]
@enxebre yep, that's fair. For what it's worth, the latest version of this PR does work fine. Thanks!

**Comment by enxebre** [comment]
cc @abhinavdahiya @wking

**Comment by derekwaynecarr** [comment]
e2e -aws failure was vpc limit.

/test e2e-aws
